app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1547,33 +1567,37 @@ class BrowserTabViewModel(
         }
     }
 
-    override fun onSavedSiteEdited(savedSite: SavedSite) {
-        when (savedSite) {
-            is SavedSite.Bookmark -> {
-                viewModelScope.launch(dispatchers.io()) {
-                    editBookmark(savedSite)
-                }
-            }
-            is SavedSite.Favorite -> {
-                viewModelScope.launch(dispatchers.io()) {
-                    editFavorite(savedSite)
-                }
-            }
+    override fun onSavedSiteFavoriteEdited(favorite: SavedSite.Favorite) {
+        viewModelScope.launch(dispatchers.io()) {
+            editFavorite(favorite)
         }
     }
 
-    fun onEditSavedSiteRequested(savedSite: SavedSite) {
-        command.value = ShowEditSavedSiteDialog(savedSite)
+    override fun onSavedSiteBookmarkEdited(bookmark: SavedSite.Bookmark, bookmarkFolder: BookmarkFolder) {
+        viewModelScope.launch(dispatchers.io()) {
+            editBookmark(bookmark, bookmarkFolder)
+        }
+    }
+
+    fun onEditSavedSiteRequested(savedSite: SavedSite, bookmarkFolder: BookmarkFolder?) {
+        command.value = ShowEditSavedSiteDialog(savedSite, bookmarkFolder)
     }
 
     fun onDeleteQuickAccessItemRequested(savedSite: SavedSite) {
         command.value = DeleteSavedSiteConfirmation(savedSite)
     }
 
-    private suspend fun editBookmark(bookmark: SavedSite.Bookmark) {
+    private suspend fun editBookmark(bookmark: SavedSite.Bookmark, bookmarkFolder: BookmarkFolder) {
         withContext(dispatchers.io()) {
             bookmarksRepository.update(bookmark)
         }
+        updateBookmarkAndFavoriteState(bookmark.url)
```

Comment:
after updating the bookmark, I would retreive the bookmark folder using `getBookmarkFolder` here. That way we can remove the method param and go back to a single method + sealed class in other classes.